pydantic/_internal/_fields.py

```diff
@@ -102,6 +102,11 @@ def collect_model_fields(  # noqa: C901
     BaseModel = import_cached_base_model()
     FieldInfo_ = import_cached_field_info()
 
+    parent_fields_lookup: dict[str, FieldInfo] = {}
+    for base in reversed(bases):
+        if model_fields := getattr(base, 'model_fields', None):
+            parent_fields_lookup.update(model_fields)
+
```

Comment:
Last q - any reason to have this up here instead of down below, where the previous logic was?

Reply:
We don't "have to", but it is equivalent and avoids doing it for every annotation (hard to see because the for loop is large, but it was previously done in the `for ann_name, ann_type in type_hints.items():` loop).

Reply:
Ahh great, loop is so large I didn't get that far up haha. Just went to the bases loop.